experiment/simulation/js/main.js

```diff
@@ -1 +1,711 @@
+			ctx.closePath();
+			ctx.fill();
+			ctx.stroke();
+
+			const e1 = [this.pos[0] + this.width, this.pos[1]], e2 = [...this.pos];
+			const gradX = (e1[0] - e2[0]) / -4, gradY = 10;
+
+			ctx.beginPath();
+			ctx.moveTo(e2[0], e2[1]);
+			curvedArea(ctx, e2, -1 * gradX, -1 * gradY);
+			curvedArea(ctx, e1, gradX, gradY);
+			ctx.closePath();
+			ctx.fill();
+			ctx.stroke();
+		};
+	};
+
+	class soil {
+		constructor(height, width, radius, x, y) {
+			this.height = height;
+			this.width = width;
+			this.radius = radius;
+			this.pos = [x, y];
+			this.color = "#654321";
+		};
+
+		draw(ctx) {
+			if(this.width < 2 * this.radius) 
+			{
+				this.radius = this.width / 2;
+			}
+
+			if(this.height < 2 * this.radius) 
+			{
+				this.radius = this.height / 2;
+			}
+
+			ctx.beginPath();
+			ctx.fillStyle = this.color;
+			ctx.beginPath();
+	
+			ctx.moveTo(this.pos[0] + this.radius, this.pos[1]);
+			ctx.arcTo(this.pos[0] + this.width, this.pos[1], this.pos[0] + this.width, this.pos[1] + this.height, this.radius);
+			ctx.arcTo(this.pos[0] + this.width, this.pos[1] + this.height, this.pos[0], this.pos[1] + this.height, this.radius);
+			ctx.arcTo(this.pos[0], this.pos[1] + this.height, this.pos[0], this.pos[1], this.radius);
+			ctx.arcTo(this.pos[0], this.pos[1], this.pos[0] + this.width, this.pos[1], this.radius);
+			ctx.closePath();
+			ctx.fill();
+		};
+	};
+
+	class soilPat {
+		constructor(height, width, angle, x, y) {
+			this.width = width; 
+			this.height = height;
+			this.angle = angle;
+			this.pos = [x, y];
+			this.grooveDims = [0, 0];
+		};
+
+		draw(ctx) {
+			ctx.fillStyle = "#b86d29";
+			const e1 = [this.pos[0] + this.width, this.pos[1] + this.height], e2 = [...this.pos], gradX = (e1[0] - e2[0]) / 2, gradY = (e1[1] - e2[1]) / 4;
+			const inter = [((10 - this.grooveDims[0]) * e1[0] + this.grooveDims[0] * e2[0]) / 10, ((10 - this.grooveDims[0]) * e1[1] + this.grooveDims[0] * e2[1]) / 10];
+
+			ctx.beginPath();
+			ctx.moveTo(e2[0], e2[1]);
+			ctx.bezierCurveTo(e2[0] - 3 * gradX, e2[1] + gradY, e1[0] - 2.5 * gradX, e1[1] + gradY, e1[0], e1[1]);
+			ctx.lineTo(e1[0] - this.grooveDims[1] / 2, e1[1]);
+			ctx.lineTo(inter[0] - this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(inter[0] + this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(e2[0], e2[1]);
+
+			ctx.moveTo(e1[0], e1[1]);
+			ctx.bezierCurveTo(e1[0] + 3.5 * gradX, e1[1] - gradY, e2[0] + 2 * gradX, e2[1] - gradY, e2[0], e2[1]);
+			ctx.lineTo(inter[0], inter[1]);
+			ctx.lineTo(inter[0] + this.grooveDims[1] / 2, inter[1]);
+			ctx.lineTo(e1[0] + this.grooveDims[1] / 2, e1[1]);
+
+			ctx.closePath();
+			ctx.fill();
+		};
+
+		groove(change) {
+			this.grooveDims[0] += change;
+
+			if(this.grooveDims[0] >= 10)
+			{
+				return 1;
+			}
+
+			return 0;
+		};
+	};
+
+	class water {
+		constructor(height, width, x, y) {
+			this.height = height;
+			this.width = width;
+			this.pos = [x, y];
+		};
+
+		draw(ctx) {
+			ctx.fillStyle = "#1ca3ec";
```

Comment:
Can we create a mnemonic const at the top of this file with this color code? So we have some idea about which color is used here or maybe use a separate colors.js file creating an enum from colornames to colorcodes. The same file can be used across main.js and plastic.js